esprima.js

```diff
@@ -187,6 +188,7 @@
         IllegalContinue: 'Illegal continue statement',
         IllegalBreak: 'Illegal break statement',
         IllegalReturn: 'Illegal return statement',
+        IllegalYield: 'Illegal yield expression',
```

Comment:
I don't think V8 has this error message yet. We may stick with the generic "Unexpected token..." until one is available.